slackbot/slack_bot.py

```diff
@@ -114,15 +114,21 @@ def replyto(msg, reply, username=workerid, broadcast=False):
 
 def shut_down_clusters():
     cluster_size = get_variable('cluster_target_size', deserialize_json=True)
-    for k in cluster_size:
-        cluster_size[k] = 0
-    set_variable("cluster_target_size", cluster_size, serialize_json=True)
+    set_variable("cluster_target_size", {k: 0 for k in cluster_size}, serialize_json=True)
+
+    min_size = get_variable('cluster_min_size', deserialize_json=True)
+    y2k = datetime(2000, 1, 1).isoformat()
```

Comment:
Since the y2k placeholder have been used in the gapi changes as well, I think it is better to abstract the trick to separate functions and give them more meaningful names to hide the detail, like init_timestamp, check_timestamp_expired etc.

Reply:
I had similar thoughts for this as the first comment -- I wasn't sure if you'd want a separate module for this. Moving `airflow_api.py` to `common` seems like a good move here as well.